pkg/util/error.go

```diff
@@ -44,10 +45,28 @@ func IsNotFound(err error) bool {
 	return false
 }
 
-func IsConnectionRefusedOrUnauthorized(err error) bool {
+func IsConnectionOrHostError(err error) bool {
```

Comment:
please rename this function to express it's for determining connection errors, something like `IsConnectionRelatedError`

Reply:
Please also think how to test this scenario, you can think about have it as e2e test, to simulate the situation have unauthorized error, no such host error. check the existing e2e how it simulate connection refused error by delete skr cluster [here](https://github.com/kyma-project/lifecycle-manager/blob/b8e3fec0e7da1d1f493b99203a87379994a89f8c/tests/e2e/deletion_test.go#L79)

Reply:
unit test only this function is pointless, basically what you do here is mock a fake connection error, then test if it return true or false. But first we don't know if this is the same as the actual connection error, more critically, what the consequence this error brings is not tested.

So what's expected to be tested here: when this connection error happens, we need to verify this [cache evict](https://github.com/kyma-project/lifecycle-manager/blob/b2fe4ec711869034bba7c77ffdf7f3f2791d2ec1/internal/declarative/v2/reconciler.go#L152-L155) and reset new valid remote client cache works as expected. I‘m fine if you can design an integration test for this scenario, but an e2e test is much more straightforward in this case.

please also remove this [block](https://github.com/kyma-project/lifecycle-manager/blob/b2fe4ec711869034bba7c77ffdf7f3f2791d2ec1/internal/declarative/v2/reconciler.go#L457-L461), this event does not bring too much value for us.

Reply:
Hmm, maybe this could be a separate issue because this also handles the testing of connection refused errors which is an existing functionality before this CR. plus we can get the feature merged faster that way. either way, it's your call

Reply:
Let's follow the engineering standard on how to properly do bug fixing/new features. Ideally, all features should be tested before being accepted, no exceptions. If you saw some function without a test covered, that's not something you should follow with, instead, as you suggest, create a followup issue or tested immediately, but the followup issue should only accepted when this issue is urgent with time pressure and approved by the manual fix as a compromise but ideally should avoid it to postpone the test, for example, we have this [issue](https://github.com/kyma-project/lifecycle-manager/issues/1310) to address that unauthorized connection error.

As I said, the connection refused errors have been already covered in the e2e test [here](https://github.com/kyma-project/lifecycle-manager/blob/b8e3fec0e7da1d1f493b99203a87379994a89f8c/tests/e2e/deletion_test.go#L79).

For enterprise software, seeking speed is not our goal, especially when the component has been running in the production environment. Instead, the code quality and making sure bug free is the first priority.

Reply:
Wrt. testing: maybe it is possible to re-use what is done [here](https://github.com/kyma-project/lifecycle-manager/blob/b8e3fec0e7da1d1f493b99203a87379994a89f8c/tests/e2e/utils_test.go#L100-L113) (called in `InitEmptyKymaBeforeAll`), but instead of replacing localhost with the k3d internal host, we replace it with some not reachable dummy value.